storm-core/src/jvm/backtype/storm/utils/TransferDrainer.java

```diff
@@ -26,88 +26,47 @@
 
 public class TransferDrainer {
 
-  private HashMap<String, ArrayList<ArrayList<TaskMessage>>> bundles = new HashMap();
-  
-  public void add(HashMap<String, ArrayList<TaskMessage>> workerTupleSetMap) {
-    for (String key : workerTupleSetMap.keySet()) {
-      
-      ArrayList<ArrayList<TaskMessage>> bundle = bundles.get(key);
-      if (null == bundle) {
-        bundle = new ArrayList<ArrayList<TaskMessage>>();
-        bundles.put(key, bundle);
-      }
-      
-      ArrayList tupleSet = workerTupleSetMap.get(key);
-      if (null != tupleSet && tupleSet.size() > 0) {
-        bundle.add(tupleSet);
-      }
-    } 
+  private ArrayList<TaskMessage> buffer = new ArrayList<TaskMessage>();
+
+  public void addAll(ArrayList<TaskMessage> tuples) {
+    buffer.addAll(tuples);
```

Comment:
I know you mentioned this in a comment on the former pull request, but since this is just a list do you want to remove `buffer`, `addAll()`, and `clear()` from this class and instead pass an ArrayList to the `send` method?